packages/enzyme/src/ShallowWrapper.js

```diff
@@ -195,6 +193,15 @@ class ShallowWrapper {
       privateSetNodes(this, nodes);
     }
     privateSet(this, OPTIONS, root ? root[OPTIONS] : options);
+
+    if (!instance) {
+      ({ instance } = this[RENDERER].getNode());
+    }
+    // Ensure to call componentDidUpdate when instance.setState is called
+    if (instance && lifecycles.componentDidUpdate.onSetState && !instance[SET_STATE]) {
```

Comment:
Nits: I should have overridden `setState` only in `options.disableLifecycleMethods` is false.

Reply:
i'll add a test for that.

Reply:
actually, can you write one? :-D i'm not sure where to put it.

or is this just an optimization?

Reply:
https://github.com/airbnb/enzyme/pull/1763/commits/a05d4cb5dcd84dfd0ba3c8f074641b173e85242f is the right place. It is unnecessary when `options. disableLifecycleMethods` is `true`.